### PR TITLE
Fix ijar failing on empty jars.

### PR DESCRIPTION
**Pull request by comius**
The problem happened when all the files are filtered out. In such case a dummy file is created, but it was not accounted for in the size estimation.
Fixes https://github.com/bazelbuild/bazel/issues/10162

### CODE REVIEW

File: third_party/ijar/ijar.cc [hunk 1]
```diff
@@ -368,9 +370,12 @@ static void OpenFilesAndProcessJar(const char *file_out, const char *file_in,
             strerror(errno));
     abort();
   }
-  u8 output_length =
-      in->CalculateOutputLength() +
-      EstimateManifestOutputSize(target_label, injecting_rule_kind);
+  u8 output_length = in->CalculateOutputLength();
+  if (output_length < 103ull + DUMMY_PATH_LENGTH) {
```
Comment (cushon) [inline third_party/ijar/ijar.cc:374]: Where does `103` come from?

File: third_party/ijar/ijar.cc [hunk 2]
```diff
@@ -368,9 +370,12 @@ static void OpenFilesAndProcessJar(const char *file_out, const char *file_in,
             strerror(errno));
     abort();
   }
-  u8 output_length =
-      in->CalculateOutputLength() +
```
Comment (cushon) [inline third_party/ijar/ijar.cc:372]: I wonder of `CalculateOutputLength` should be handling this instead of fixing up the size later?

**Review by cushon — APPROVED** [review]
Consider adding a test?

I also commented in the bug--it's not clear if this is something that happened in a real build, it should be rare to see completely empty jar files that don't even contain a manifest.

**Review by ahmadasiri-24 — COMMENTED** [review]
third_party/ijar/ijar.cc

File: third_party/ijar/ijar.cc (hunk 2 above)
Comment (comius) [reply third_party/ijar/ijar.cc:372]: > Consider adding a test?

Done

> I also commented in the bug--it's not clear if this is something that happened in a real build, it should be rare to see completely empty jar files that don't even contain a manifest.

True that.

Comment (comius) [reply third_party/ijar/ijar.cc:372]: `CalculateOutputLength` is part of `zip.cc` and has no knowledge that we're adding `dummy` file.

File: third_party/ijar/ijar.cc (hunk 1 above)
Comment (comius) [reply third_party/ijar/ijar.cc:374]: From measurement, error message tells 108 bytes are needed. "dummy" appears twice in the output, so 98+2*strlen("dummy") should work even if somebody changes "dummy". (I fixed 103).

Somehow I cannot get the same number by adding together numbers used EstimateManifestOutputSize. And I consider reverse engineering zip code and figuring out header sizes a bit too much effort for this fix.

**Pull request closed without merging**